### PR TITLE
Add configBranch to app patch and registration

### PR DESCRIPTION
**Pull request by nilsgstrabo**

### CODE REVIEW

File: api/applications/applications_handler.go [hunk 1]
```diff
@@ -270,6 +271,19 @@ func (ah ApplicationHandler) ModifyRegistrationDetails(appName string, patchRequ
 		runUpdate = true
 	}
 
+	if patchRequest.ConfigBranch != nil && *patchRequest.ConfigBranch != "" {
```
Comment (satr) [inline api/applications/applications_handler.go:274]: I suggest we start always trim spaces from requested strings

**Review by satr — CHANGES_REQUESTED** [review]
I suggest one correction - trim spaces in input request.

File: api/applications/applications_handler.go (hunk 1 above)
Comment (nilsgstrabo) [reply api/applications/applications_handler.go:274]: Agreed. I'll fix it

Comment (nilsgstrabo) [reply api/applications/applications_handler.go]: Implemented your suggestions

**Review by satr — APPROVED** [review]

**Pull request merged**